app/src/main/java/org/p2p/wallet/user/repository/UserRemoteRepository.kt

```diff
@@ -42,75 +45,87 @@ class UserRemoteRepository(
+                val response = json.get("Response")
+                if (response?.asString == "Error") {
+                    throw IllegalStateException("Cannot get rates")
                 }
+                list.forEach { symbol ->
+                    val tokenObject = json.getAsJsonObject(symbol.uppercase())
+                    if (tokenObject != null) {
+                        val price = tokenObject.getAsJsonPrimitive(USD_READABLE_SYMBOL).asBigDecimal
+                        prices.add(TokenPrice(symbol, price.scaleMedium()))
+                    }
+                }
+            }
 
-                val token = userLocalRepository.findTokenData(mintAddress) ?: return@mapNotNull null
-                val price = userLocalRepository.getPriceByToken(token.symbol)
-                TokenConverter.fromNetwork(it, token, price)
+        return prices
+    }
+
+    private suspend fun mapAccountsToTokens(publicKey: String, accounts: List<Account>): List<Token.Active> {
+        val tokens = accounts.mapNotNull {
+            val mintAddress = it.account.data.parsed.info.mint
+
+            if (mintAddress == REN_BTC_DEVNET_MINT) {
+                return@mapNotNull mapDevnetRenBTC(it)
+            }
+
+            if (mintAddress == WRAPPED_SOL_MINT) {
+                // Hiding Wrapped Sol account because we are adding native SOL lower
+                return@mapNotNull null
             }
-            .toMutableList()
+
+            val token = userLocalRepository.findTokenData(mintAddress) ?: return@mapNotNull null
+            val price = userLocalRepository.getPriceByToken(token.symbol)
+            TokenConverter.fromNetwork(it, token, price)
+        }
 
         /*
          * Assuming that SOL is our default token, creating it manually
          * */
+        return addSolToken(publicKey, tokens)
+    }
+
+    private suspend fun addSolToken(publicKey: String, tokens: List<Token.Active>): List<Token.Active> {
         val solBalance = rpcBalanceRepository.getBalance(publicKey)
-        val tokenData = userLocalRepository.findTokenData(WRAPPED_SOL_MINT) ?: return@withContext result
+        val tokenData = userLocalRepository.findTokenData(WRAPPED_SOL_MINT) ?: return tokens
         val solPrice = userLocalRepository.getPriceByToken(tokenData.symbol)
-        val token = Token.createSOL(
+        val solToken = Token.createSOL(
             publicKey = publicKey,
             tokenData = tokenData,
             amount = solBalance,
             exchangeRate = solPrice?.getScaledValue()
         )
-        result.add(0, token)
-        return@withContext result
-    }
 
-    override suspend fun getRate(sourceSymbol: String, destinationSymbol: String): TokenPrice? {
-        val json = compareApi.getPrice(sourceSymbol, destinationSymbol)
-        val price = json.getAsJsonPrimitive(destinationSymbol)?.asBigDecimal
-        return price?.let { TokenPrice(sourceSymbol, price.scaleMedium()) }
+        return tokens + solToken
```

Comment:
Sol should be first in the list, therefore let's do below:
`solToken + tokens`

Reply:
`TokenComparator` do that work so current order doesn't matter

Reply:
Yes, it does, but it will not make additional replacing if SOL is in correct position which reduces the sorting time

Reply:
You right